docs/contributors/storybook.md

```diff
@@ -1,11 +1,26 @@
-# Storybook
+# Storybook & Components
 
 This repo includes [Storybook](https://storybook.js.org) tooling so we can test and develop components in isolation.
 
 The storybook is automatically built and published to [GitHub pages](https://woocommerce.github.io/woocommerce-gutenberg-products-block/) on every push to the main branch. See [travis.yml](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/main/.travis.yml) for details.
 
 https://woocommerce.github.io/woocommerce-gutenberg-products-block/
 
+## Where are our components?
+We have components in a few folders, for different contexts.
+
+- [`assets/js/base/components`](https://github.com/woocommerce/woocommerce-gutenberg-products-block/tree/main/assets/js/base/components)
+- [`assets/js/editor-components`](https://github.com/woocommerce/woocommerce-gutenberg-products-block/tree/main/assets/js/components)
+- [`assets/js/icons`](https://github.com/woocommerce/woocommerce-gutenberg-products-block/tree/main/assets/js/icons)
+
+__`base/components`__ are used in front-end code (as well as editor & admin). These components are typically general purpose (and the largest collection). These components help us build consistent interfaces across the front end (shopper) experience and elsewhere. In time, some of these components may move to a general purpose lib, such as [`@wordpress/components` in Gutenberg](https://github.com/WordPress/gutenberg/tree/master/packages/components).
+
+__`base/editor-components`__ are used in the editor UI for our blocks. These are much more specific to WooCommerce Blocks. They allow us to build a consistent and powerful UI for merchants for authoring content relating to Woo data - e.g. selecting products or product attributes.
```

Comment:
I'm a bit wary on using `editor-components` before that folder exists. I would just list `assets/js/components` for now until we rename the folder.

Related to that, if we are renaming `assets/components` to `assets/editor-components`, shouldn't we also rename `assets/hocs` to `assets/editor-hocs`? :thinking:

Reply:
> I'm a bit wary on using editor-components before that folder exists. I would just list assets/js/components for now until we rename the folder.

I agree with Albert here.

> Related to that, if we are renaming assets/components to assets/editor-components, shouldn't we also rename assets/hocs to assets/editor-hocs

Also agree, we'll have to make sure only editor specific hocs are in this folder though.

Reply:
Good spotting - I renamed it in the docs first to get feedback before making the change for real (as noted in PR description).